### PR TITLE
Cleanup model import/export with factorizer

### PR DESCRIPTION
**Pull request by glinscott**
@vondele @Sopel97 

train.py
- Adds `--enable-factorizer` option.  Default is false, which will train without factorizer.  Factorizer on or off dynamically modifies the model shape, so you can resume non-factorized nets in factorizer mode.
- Adds `--reuse-from-model` option.  Pass in a serialized model, ending in `.pt`, to resume from those weights.

serialize.py: 
- Fixes nnue -> pt, as they need to be imported without factorizer shapes.
- Supports both factorized and non-factorized pt -> nnue conversions.

### CODE REVIEW

**Comment by Sopel97** [comment]
would make sense to have the option be `--enable-factorizer` for both?

File: halfkp.py [hunk 1]
```diff
@@ -27,7 +29,7 @@ def halfkp_idx(is_white_pov: bool, king_sq: int, sq: int, p: chess.Piece):
 def get_halfkp_indices(board: chess.Board):
   # TODO - this doesn't support the factors yet.
   def piece_indices(turn):
-    indices = torch.zeros(INPUTS)
+    indices = torch.zeros(Features.inputs)
     for sq, p in board.piece_map().items():
       if p.piece_type == chess.KING:
         continue
```
Comment (Sopel97) [inline halfkp.py]: perhaps these should now be a part of `class Features:`

Comment (Sopel97) [reply halfkp.py]: also, the lines https://github.com/glinscott/nnue-pytorch/blob/7793bf6620fcd8003692d755c0e735f8ff5566de/serialize.py#L12-L32 could become a part of the factorizer

Comment (glinscott) [reply halfkp.py]: Good call, done.

**Comment by glinscott** [comment]
Switched both options to `--enable-factorizer`.

**Comment by glinscott** [comment]
Actually, ended up removing `--enable-factorizer` from serialize.py.  It's cleaner to only have it in train.py, which now adjusts the model based on the options passed into training.

File: README.md [hunk 2]
```diff
@@ -51,6 +51,11 @@ Import an existing SF NNUE network to the pytorch network format.
 python serialize.py nn.nnue converted.pt
 ```
 
+Import the network, with the factorizer enabled (for resuming training):
+```
+python serialize.py nn.nnue converted.pt --enable-factorizer
```
Comment (vondele) [inline README.md]: needs update?

Comment (glinscott) [reply README.md]: Done, thanks.

**Comment by glinscott** [comment]
Ok, I think this is ready to merge now if it looks good to both of you.

**Comment by vondele** [comment]
I've just been testing and either I'm doing something wrong, or there is an issue:
```
python serialize.py nn-c3ca321c51c9.nnue nn-c3ca321c51c9.pt
python train.py  --resume-from-model nn-c3ca321c51c9.pt --gpus 1 ../gensfen/data05/d5_128000000_1468_staticEval.binpack ../gensfen/data05/d5_128000000_1468_staticEval.binpack
```
gives a large loss: loss=0.060 whereas it should be 0.000

**Comment by glinscott** [comment]
Oops, good catch.  Pushed a fix.

**Comment by vondele** [comment]
excellent restarts from .pt now work with and without factorizer, giving the expected zero loss.
